betelgeuse.py

```diff
@@ -814,3 +814,134 @@ def test_run(
+    '--not-finished',
+    help='Indicate to the importer to set the status to "In Progress" instead '
+    'of "Finished".',
+    is_flag=True,
+)
+@click.option(
+    '--not-include-skipped',
+    help='Indicate to the importer to no include skipped tests.',
+    is_flag=True,
+)
+@click.option(
+    '--test-run-id',
+    default='test-run-{0}'.format(time.time()),
+    help='Test Run ID to be created/updated.',
+)
+@click.option(
+    '--test-run-title',
+    help='Test Run title.',
+)
+@click.argument('junit-path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('source-code-path', type=click.Path(exists=True))
+@click.argument('user')
+@click.argument('project')
+@click.argument('output-path')
+@click.pass_context
+def xml_test_run(
+        context, custom_fields, dry_run, lookup_method, not_finished,
+        not_include_skipped, test_run_id, test_run_title, junit_path,
+        source_code_path, user, project, output_path):
+    """Generate an XML suited to be importer by the test-run importer."""
+    test_run_id = re.sub(INVALID_CHARS_REGEX, '', test_run_id)
+    testsuites = ElementTree.Element('testsuites')
+    properties = ElementTree.Element('properties')
+    custom_fields = load_custom_fields(custom_fields)
+    if dry_run:
+        custom_fields['polarion-dry-run'] = 'true'
+    if not_finished:
+        custom_fields['polarion-set-testrun-finished'] = 'false'
+    if not_include_skipped:
+        custom_fields['polarion-include-skipped'] = 'false'
+    custom_fields['polarion-lookup-method'] = lookup_method
+    custom_fields['polarion-project-id'] = project
+    custom_fields['polarion-testrun-id'] = test_run_id
+    if test_run_title:
+        custom_fields['polarion-testrun-title'] = test_run_title
+    custom_fields['polarion-user-id'] = user
+    properties_names = (
+        'polarion-dry-run',
+        'polarion-include-skipped',
+        'polarion-lookup-method',
+        'polarion-project-id',
+        'polarion-set-testrun-finished',
+        'polarion-testrun-id',
+        'polarion-testrun-title',
+        'polarion-user-id',
+    )
+    for name, value in custom_fields.items():
+        if (not name.startswith('polarion-custom-') and
+                name not in properties_names):
+            name = 'polarion-custom-{}'.format(name)
+        properties.append(create_xml_property(name, value))
+    testsuites.append(properties)
+
+    testcases = {
+        generate_test_id(test): test.tokens.get('id')
+        for test in itertools.chain(
+                *testimony.get_testcases([source_code_path]).values()
+        )
+    }
+    testsuite = ElementTree.parse(junit_path).getroot()
+    if 'skips' in testsuite.attrib:
+        testsuite.attrib['skipped'] = testsuite.attrib['skips']
+        del testsuite.attrib['skips']
+    for testcase in testsuite.iterfind('testcase'):
+        if 'file' in testcase.attrib:
+            del testcase.attrib['file']
+        if 'line' in testcase.attrib:
```

Comment:
Best to comment here so that we can remember later on why we're doing this. I know the answer now but will I remember it later? :)